www/src/pages/editor/NewMineralSiteModal.tsx

```diff
@@ -0,0 +1,425 @@
+import { LocationInfo } from "../../models/mineralSite/LocationInfo";
+import { Reference, Document } from "../../models/mineralSite/Reference";
+import { GradeTonnage } from "../../models/mineralSite/GradeTonnage";
+import { CommodityStore } from "models/commodity";
+import axios from "axios";
+import { Country, StateOrProvince } from "models";
+import { useEffect, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+import { UserStore } from "models/user";
+import { MineralInventory, Measure } from "../../models/mineralSite/MineralInventory";
+import { useMemo } from "react";
+import { UnitStore } from "models/units";
+
+interface NewMineralSiteModalProps {
+  commodity: Commodity;
+  visible: boolean;
+  onClose: () => void;
+}
+interface FormValues {
+  latitude?: number;
+  longitude?: number;
+  country?: string;
+  stateOrProvince?: string;
+  name: string;
+  refDoc: string;
+  refComment?: string;
+  sourceType: string;
+  depositType: string;
+  depositTypeConfidence: number;
+  tonnage?: number;
+  grade?: number;
+  gradeUnit?: string;
+  tonnageUnit?: string;
+}
+
+export const NewMineralSiteModal: React.FC<NewMineralSiteModalProps> = ({ commodity, visible, onClose }) => {
+  const { mineralSiteStore, dedupMineralSiteStore, userStore, commodityStore, countryStore, stateOrProvinceStore, depositTypeStore, unitStore } = useStores();
+  const [form] = Form.useForm();
+  const [recordId, setRecordId] = useState("");
+  const [selectedSourceType, setSelectedSourceType] = useState<string | null>(null);
+
+  const commodityOptions = useMemo(() => {
+    return commodityStore.list.map((comm) => ({
+      value: comm.uri,
+      label: comm.name,
+    }));
+  }, [commodityStore.records]);
+  const currentUserUrl = userStore.getCurrentUser()!.url;
+  const countryOptions = useMemo(() => {
+    return countryStore.list.map((country) => ({
+      value: country.uri,
+      label: country.name,
+    }));
+  }, [countryStore.records]);
+
+  const stateOptions = useMemo(() => {
+    return stateOrProvinceStore.list.map((state) => ({
+      value: state.id,
+      label: state.name,
+    }));
+  }, [stateOrProvinceStore.records]);
+
+  const depositTypeOptions = useMemo(() => {
+    return depositTypeStore.list.map((type) => ({
+      value: type.id,
+      label: type.name,
+    }));
+  }, [depositTypeStore.records]);
+
+  const unitOptions = useMemo(() => {
+    return unitStore.list.map((unit) => ({
+      value: unit.id,
+      label: unit.name,
+    }));
+  }, [unitStore.records]);
+
+  const handleSourceTypeChange = (e: RadioChangeEvent) => {
+    const value = e.target.value;
+    setSelectedSourceType(value);
+
+    const refDocUrl = form.getFieldValue("refDoc");
+    if (value === "unpublished") {
+      form.setFieldsValue({ sourceId: `unpublished::${currentUserUrl}` });
+    } else if (refDocUrl) {
+      form.setFieldsValue({ sourceId: `${value}::${refDocUrl}` });
+    }
```

Comment:
what happens when refDocURL is "false"?

Reply:
if (value === "unpublished") {
      form.setFieldsValue({ sourceId: `unpublished::${currentUserUrl}` });
    }
    if (refDocUrl) {
      form.setFieldsValue({ sourceId: `${value}::${refDocUrl}` });
    }

Reply:
is this okay?

Reply:
I don't think your changes address the question when refDocUrl is false